R/narr_data_functions.R

```diff
@@ -80,48 +53,9 @@ get_narr_data <- function(d,
     stop("input dataframe must have a column called 'end_date'")
   }
 
-  d <- split(d, d$narr_cell)
-
-  return(purrr::map_dfr(d, ~ read_narr_fst_join(.x, narr_variables, narr_fst_filepath = narr_fst_filepath)))
-}
-
-#' download narr.fst file to application specific directory so that it can be shared across R sessions and projects
-#' @export
-download_narr_fst <- function() {
-  narr_fl_appdir <- fs::path(rappdirs::user_data_dir("addNarrData"), "narr.fst")
-
-  if (!fs::file_exists(narr_fl_appdir)) {
-    cli::cli_alert_info("This package requires a local copy of s3://geomarker/narr/narr.fst in order to lookup NARR values; it is > 22 GB in size and will be downloaded to ",
-                        narr_fl_appdir, " so it can be shared across R sessions and projects.")
-    ans <- readline("Do you want to download this now (Y/n)? ")
-    if (!ans %in% c("", "y", "Y")) stop("aborted", call. = FALSE)
-
-    fs::dir_create(rappdirs::user_data_dir("addNarrdata"))
-    withr::with_envvar(new = c(
-      "AWS_ACCESS_KEY_ID" = NA,
-      "AWS_SECRET_ACCESS_KEY" = NA
-    ), {
-      s3::s3_get('s3://geomarker/narr/narr.fst', download_folder = rappdirs::user_data_dir("addNarrdata"))
-    })
-  }
-}
-
-
-#' checks for and returns filepath to narr.fst file in application-specific site data directory or current working directory
-#' if not found, fails with suggestion to run download_narr_fst()
-narr_fst <- function() {
-  narr_fl_appdir <- fs::path(rappdirs::user_data_dir("addNarrData/geomarker/narr"), "narr.fst")
-  narr_fl_wd <- fs::path(getwd(), "/geomarker/narr/narr.fst")
-
-  if (file.exists(narr_fl_wd)) {
-    message("using narr.fst file at ", narr_fl_wd)
-    return(invisible(narr_fl_wd))
-  }
-
-  if (file.exists(narr_fl_appdir)) {
-    message("using narr.fst file at ", narr_fl_appdir)
-    return(invisible(narr_fl_appdir))
-  }
+  d$narr_chunk <- d$narr_cell %/% 10000
+  d <- dht::expand_dates(d, by = 'day')
+  d_split <- split(d, d$narr_chunk)
 
-  stop("narr.fst file not found at\n    ", narr_fl_appdir, "\n    or at\n    ", narr_fl_wd, "\n please call download_narr_fst() to download this file", call. = FALSE)
+  return(purrr::map_dfr(d_split, ~ download_join_chunks(.x, narr_variables)))
```

Comment:
Let's silence all messages from `s3` and above (also make sure to not require a confirmation from the user when using `s3::s3_get_files()`).  

Instead, we will add a simple message progress bar here that just announces each time it processes a chunk. Or use something like we did in `s3` to update the user: https://github.com/geomarker-io/s3/blob/ed7530a213139c92d8e5984661f9158afa07e174/R/s3_get_files.R#L94-L122